### PR TITLE
integration test fixes for 4.18

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR...

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: test/integration/smoke/test_deploy_vm_extra_config_data.py [hunk 1]
```diff
@@ -63,34 +63,27 @@ def setUpClass(cls):
...
     def tearDownClass(cls):
-        try:
-            cls.apiclient = super(TestAddConfigtoDeployVM, cls).getClsTestClient().getApiClient()
-            cls.reset_hosts_hugepages()
-            # Clean up, terminate the created templates
-            cleanup_resources(cls.apiclient, cls.cleanup)
-
-        except Exception as e:
-            raise Exception("Warning: Exception during cleanup : %s" % e)
+        super(TestAddConfigtoDeployVM, cls).tearDownClass()
```
Comment (DaanHoogland) [inline test/integration/smoke/test_deploy_vm_extra_config_data.py:86]: @weizhouapache, stolen from #7186. We are working on the same stuff ...
```suggestion
        super(TestAddConfigtoDeployVM, cls).tearDownClass()
        cls.reset_hosts_hugepages()
```

File: test/integration/smoke/test_deploy_vm_extra_config_data.py [hunk 2]
```diff
@@ -130,6 +124,10 @@ def setUp(self):
         updateConfigurationResponse = self.apiclient.updateConfiguration(updateConfigurationCmd)
         self.debug("updated the parameter %s with value %s" % (
             updateConfigurationResponse.name, updateConfigurationResponse.value))
+        self.cleanup = []
```
Comment (weizhouapache) [inline test/integration/smoke/test_deploy_vm_extra_config_data.py:127]: it is empty by default, right ?

**Review by weizhouapache — APPROVED** [review]
code lgtm

File: test/integration/smoke/test_deploy_vm_extra_config_data.py (hunk 2 above)
Comment (DaanHoogland) [reply test/integration/smoke/test_deploy_vm_extra_config_data.py:127]: yes, it is in this file only filled by deploy_vm()

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test matrix

**Comment by weizhouapache** [comment]
@DaanHoogland 
you need to create vm using VirtualMachine.create, or roll back all changes on deploy/destroy vm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@weizhouapache I reverted changes on test_vm_deployment_planner as the failures seem intemitted, and I don't see a reason. I reverted most changes on test_deploy_vm_extra_config_data, but added the possibility to call on the marvin object directly.

**Review by weizhouapache — APPROVED** [review]

**Comment by weizhouapache** [comment]
> @weizhouapache I reverted changes on test_vm_deployment_planner as the failures seem intemitted, and I don't see a reason. I reverted most changes on test_deploy_vm_extra_config_data, but added the possibility to call on the marvin object directly.

The vm, which is created by deployVirtualMachine.deployVirtualMachineCmd, cannot be cleaned up by tearDown class.
need to use VirtualMachine.create, like other tests, then it can be cleaned up in tearDown.

**Comment by DaanHoogland** [comment]
> > @weizhouapache I reverted changes on test_vm_deployment_planner as the failures seem intemitted, and I don't see a reason. I reverted most changes on test_deploy_vm_extra_config_data, but added the possibility to call on the marvin object directly.
> 
> The vm, which is created by deployVirtualMachine.deployVirtualMachineCmd, cannot be cleaned up by tearDown class. need to use VirtualMachine.create, like other tests, then it can be cleaned up in tearDown.

Yes, I realise that. I think that is a big change now, it got me some request signing errors trying. I want to merge as is.

**Review by nvazquez — APPROVED** [review]
Code LGTM, can we run a final round of tests?

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**